File: 1248. Count Number of Nice Subarrays/numberOfSubarrays.py
class numberOfSubarrays:
    def solution(self, nums, k: int) -> int:
        even = [0]
        res = 0
        for i in nums:
            if i & 1:
                even.append(0)
            else:
                even[-1] = even[-1] + 1
        for i in range(k, len(even)):
            res = res + (even[i - k] + 1) * (even[i] + 1)
        return res


obj = numberOfSubarrays()
nums = [1,1,2,1,1]
k = 3
res = obj.solution(nums, k)
print(res)

File: 1248. Count Number of Nice Subarrays/numberOfSubarrays.java
class numberOfSubarrays {
    public int solution(int[] nums, int k) {
        int left = 0;
        int right = 0;
        int oddCnt = 0;
        int res = 0;
        while (right < nums.length) {
            if ((nums[right++] & 1) == 1) {
                oddCnt ++;
            }
            if (oddCnt == k) {
                int temp = right;
                while (right < nums.length && (nums[right] & 1) == 0) {
                    right ++;
                }
                int rightEvenCnt = right - temp;
                int leftEvenCnt = 0;
                while((nums[left] & 1) == 0) {
                    leftEvenCnt ++;
                    left ++;
                }
                res = res + (leftEvenCnt + 1) * (rightEvenCnt + 1);
                left ++;
                oddCnt --;
            }
        }
        return res;
    }
}